subprocess/cmd.go

```diff
@@ -0,0 +1,110 @@
+package subprocess
+
+import (
+	"io"
+	"os"
+	"os/exec"
+)
+
+// Thin wrapper around exec.Cmd. Takes care of pipe shutdown by
+// keeping an internal reference to any created pipes. Whenever
+// Cmd.Wait() is called, all created pipes are closed.
+type SafeCmd struct {
+	Path string
+	Args []string
+	Env  []string
+	Dir  string
+
+	Stdin  io.Reader
+	Stdout io.Writer
+	Stderr io.Writer
+
+	Process *os.Process
+
+	cmd        *exec.Cmd
```

Comment:
I spoke a bit too soon. I need to be able to override the `*Pipe()` functions and call back to the original `exec.Cmd` function to do so. Fortunately, I can still include a reference to the original `exec.Cmd` struct while embedding. Cuts the code down quite a bit.